azure/compute/src/main/java/org/jclouds/azurecompute/compute/internal/AzureTemplateBuilderImpl.java

```diff
@@ -0,0 +1,140 @@
+          * It seems preferable to take images whose fields are *not* null, ie nullsFirst;
+          * and to use something like the AlphaNum Algorithm then take the last
+          * (so "Ubuntu 13.04" would be preferred over "Ubuntu 9.10").
+          * However not changing it now as people may be surprised if the images they get back start changing.
+          */
+            return ComparisonChain.start()
+                    .compare(left.getName(), right.getName(), Ordering.<String>natural().nullsLast())
+                    .compare(left.getVersion(), right.getVersion(), Ordering.<String> natural().nullsLast())
+                    .compare(left.getDescription(), right.getDescription(), Ordering.<String> natural().nullsLast())
+                    .compare(left.getOperatingSystem().getName(), right.getOperatingSystem().getName(),//
+                            Ordering.<String> natural().nullsLast())
+                    .compare(toLong(left.getOperatingSystem().getVersion()), toLong(right.getOperatingSystem().getVersion()),//
+                            Ordering.<Long> natural().nullsLast())
+                    .compare(left.getOperatingSystem().getDescription(), right.getOperatingSystem().getDescription(),//
+                            Ordering.<String> natural().nullsLast())
+                    .compare(left.getOperatingSystem().getArch(), right.getOperatingSystem().getArch(),//
+                            Ordering.<String> natural().nullsLast())
+                    .compare(left.getStatus().toString(), right.getStatus().toString(),
+                            Ordering.<String>natural().nullsLast())
+                    .result();
+        }
+    };
+
+    @Inject
+    protected AzureTemplateBuilderImpl(@Memoized Supplier<Set<? extends Location>> locations,
+                                       ImageCacheSupplier images, @Memoized Supplier<Set<? extends Hardware>> sizes,
+                                        Supplier<Location> defaultLocation, @Named("DEFAULT") Provider<TemplateOptions> optionsProvider,
+                                        @Named("DEFAULT") Provider<TemplateBuilder> defaultTemplateProvider, GetImageStrategy getImageStrategy) {
+        super(locations, images, sizes, defaultLocation, optionsProvider, defaultTemplateProvider, getImageStrategy);
+        imageChooser = imageChooser(IMAGE_ORDERING);
+    }
+
+    Function<Iterable<? extends Image>, Image> imageChooser(final Ordering<Image> ordering) {
+        return new Function<Iterable<? extends Image>, Image>() {
+            @Override
+            public Image apply(Iterable<? extends Image> input) {
+                // filter private images
+                Predicate<Image> statusPredicate = new Predicate<Image>() {
+
+                    @Override
+                    public boolean apply(Image image) {
+                        return image.getStatus() == Image.Status.AVAILABLE;
+                    }
+                };
```

Comment:
Is this predicate needed? According to [your image transformation function](https://github.com/MSOpenTech/jclouds-labs/blob/azure/azure/compute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java#L83) images are always available. If that's not true, then that function should be improved to populate the right status.